src/slider/interfaces.ts

```diff
@@ -0,0 +1,71 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+import { BaseComponentProps } from '../internal/base-component';
+import { FormFieldValidationControlProps } from '../internal/context/form-field-context';
+import { NonCancelableEventHandler } from '../internal/events';
+
+export interface SliderProps extends BaseComponentProps, FormFieldValidationControlProps {
+  /**
+   * Indicates the current value.
+   */
+  value?: number;
+
+  /**
+   * Indicates the minimum value.
+   */
+  min: number;
+
+  /**
+   * Indicates the maximum value.
+   */
+  max: number;
+
+  /**
+   * Formats the values. This will format both the labels and the tooltip.
+   */
+  valueFormatter?: (value: number) => string;
+
+  /**
+   * Called when the user selects a value.
+   * The event `detail` contains the current `value`.
+   */
+  onChange?: NonCancelableEventHandler<SliderProps.ChangeDetail>;
+
+  /**
+   * How big the step size is.
+   */
+  step?: number;
+
+  /**
+   * Hides the colored fill line, so only the handle is visible.
+   */
+  hideFillLine?: boolean;
+
+  /**
+   * Show the tick marks along the axis. Use with stepped sliders, except in extreme cases.
+   */
+  tickMarks?: boolean;
+
+  /**
+   * Whether or not the slider is disabled.
+   */
+  disabled?: boolean;
```

Comment:
Should we support readOnly? See sixtAeSZSxjR

Reply:
Good question. If we should, I want to say let's keep that separate and do that alongside the rest of the readOnly updates.